lib/output-marshaller.js

```diff
@@ -0,0 +1,40 @@
+const {Transform} = require('stream');
+const logger = require('util').debuglog('riff');
+const RiffError = require('./riff-error');
+const {canMarshall, determineContentTypes, marshaller} = require('./content-negotiation');
+
+module.exports = class OutputMarshaller extends Transform {
+
+    constructor(index, contentType, options) {
+        super(options);
+        if (index < 0) {
+            throw new RiffError('error-output-index-invalid', `invalid output index: ${index}`);
+        }
+        const acceptedContentType = determineContentTypes(contentType).accept;
+        if (!canMarshall(acceptedContentType)) {
+            throw new RiffError('error-output-content-type-unsupported', `unrecognized output #${index}'s content-type ${contentType}`);
+        }
+        this._index = index;
+        this._acceptedContentType = acceptedContentType;
+        this._marshallerFunction = marshaller(this._acceptedContentType);
+    }
+
+    _transform(value, _, callback) {
+        let payload;
+        try {
+            payload = this._marshallerFunction(value);
```

Comment:
what if the function returns a Message?

Reply:
Thanks, I missed that part in the current version.

Reply:
thanks, I missed that part when reading the current invoker's code.

Reply:
Now fixed.